### DISCUSSION TITLE
Should FastifySchema.response be renamed to reply?

### DISCUSSION THREAD

**Discussion by shwao** [fastify/fastify#3415] [General]
The field for the reply schema is called `response` even though its `reply` everywhere else. Should it be considered to rename it?

I know this would be a breaking change and only suitable for a major release, but its something i stumble on every time.

When you create an interface for the schema its also called `Reply`. For example:

```ts
export interface CreateOrderInterface
{
	Body: CreateOrderBody;
	Reply: CreateOrderReply;
}

export const createOrderSchema: FastifySchema = {
	body: createOrderBody,
	response: {
		200: createOrderReply,
	},
};
```

This was also confused in the documentation a while back. I made a PR for that. `Response` was used when it should have been `Reply`, the author probably was confused because its `response` in the `FastifySchema` type.

The interface i am talking about is here:
https://github.com/fastify/fastify/blob/9e4b7cc12c4beff53d08d6512d6347920a1f9842/types/schema.d.ts#L7-L13

I think it should be changed but i didn't want to directly create an issue. Maybe there is good reason for it? I don't see it but i think it should be discussed.

**Comment by Eomm** [comment]
The `Reply` object is a Fastify component that enhances the `http.serverResponse` object.

The `schema.response.200` field refers to the HTTP response and its status code, so I don't think those two entities are related.

**Reply by shwao** [reply]
I was not talking about the `FastifyReply` object but the `Reply` in the `RouteGeneric`  interface here:

https://github.com/fastify/fastify/blob/9e4b7cc12c4beff53d08d6512d6347920a1f9842/types/reply.d.ts#L32

You use `schema.response` to set the schema for the response but have to use `Reply` to set the types for the response. Maybe the `RouteGeneric` should/could be changed?

**Comment by shwao** [comment]
I should have made it clear that i am talking about the naming in the `RouteGeneric` interface.

https://github.com/fastify/fastify/blob/9e4b7cc12c4beff53d08d6512d6347920a1f9842/types/reply.d.ts#L32
